tokio/src/fs/file.rs

```diff
@@ -27,9 +27,11 @@ use std::task::Poll::*;
 /// it was opened with. Files also implement Seek to alter the logical cursor
 /// that the file contains internally.
 ///
-/// Files are automatically closed when they go out of scope.
+/// Files are automatically closed when they go out of scope, but you should always call
+/// [`flush`][flush] to ensure changes are written to disk.
```

Comment:
Actually, thinking a bit more about this, should you _always_ call flush?

Reply:
in `std`, `flush` is a no-op. In Tokio, it waits for the previous write to reach the thread pool.

Either way, `flush` does not ensure that data is written to disk. `sync` does this.